msteams-platform/messaging-extensions/high-quality-message-extension.md

```diff
@@ -0,0 +1,511 @@
+
+**Parameter description example:**
+
+```json
+"parameters": [
+  {
+    "name": "SurveyName",
+    "title": "Name of Survey",
+    "description": "Send survey name or related keyword",
+    "inputType": "text"
+  },
+  {
+    "name": "Tags",
+    "title": "Tags",
+    "description": "Send product name or keywords related pertaining to a question",
+    "inputType": "text"
+  },
+  {
+    "name": "ResponseNumber",
+    "title": "Response number",
+    "description": "Send number of responses received for a survey. Output: Number",
+    "inputType": "text"
+  }
+]
+```
+
+# [CRM](#tab/crm)
+
+Basic search: Fetch me qualified leads. </br>
+Advanced search: Fetch qualified leads for which quotes are pending from last seven days.
+
+**Parameter description example:**
+
+```json
+"parameters": [
+  {
+    "name": "TypeofLeads",
+    "title": "Type of Leads",
+    "description": "Send what types of leads user is looking for. Acceptable fields are: Qualified, Unqualified and New.",
+    "inputType": "text"
+  },
+  {
+    "name": "Status",
+    "title": "Status",
+    "description": "Send status of leads. Acceptable fields are: Pending, Quote Given and Quote Rejected.",
+    "inputType": "text"
+  },
+  {
+    "name": "Time",
+    "title": "Time",
+    "description": "Send number of days for which you need status of leads for. Output: Number",
+    "inputType": "text"
+  }
+]
+```
+
+# [General](#tab/general)
+
+Basic search: Find stocks in NASDAQ.</br>
+Advanced search: Find top 10 stocks in NASDAQ with P/E less than 30 and P/B less than 2.
+
+**Parameter description example:**
+
+```json
+"parameters": [
+  {
+    "name": "StockIndex",
+    "title": "Stock Index",
+    "description": "Name of index in which user wants to find stocks",
+    "inputType": "text"
+  },
+  {
+    "name": "NumberofStocks",
+    "title": "Ranked Number of Stocks",
+    "description": "Provide number of stocks in ranked order. Output format: Top:<Number of stocks or bottom:<Number of stocks>",
+    "inputType": "text"
+  },
+  {
+    "name": "P/B",
+    "title": "Price to Book Ratio",
+    "description": "Send P/B or Price to book ratio of a stock. Output format: >x.xx or <x.xx",
+    "inputType": "text"
+  },
+  {
+    "name": "P/E",
+    "title": "Price to Earnings Ratio",
+    "description": "Send P/E or Price to Earnings ratio of a stock with comparison. Output format: >x.xx or <x.xx",
+    "inputType": "text"
+  }
+]
+```
+
+---
+
+## Compound Utterances
+
+> [!NOTE]
+> Search through task module isn't supported in Copilot.
+
+For Copilot, a search-based message extension must support compound utterances to perform deep retrieval of accurate information. To enable compound utterances, we recommend that you expand the scope of search to handle two or more search parameters simultaneously by enabling multi-parameter support in app manifest (previously called Teams app manifest).
```

Comment:
I'm not sure if multi-param support in Copilot will be ready by 11/15. Could you confirm w/PM?

Reply:
Received an update from the SME that Multiparam is supported WW in copilot.